### PR TITLE
add interface for non-watch metakv

### PR DESCRIPTION
**Pull request by yiwangdr**
Removing watch() from metakv interface and creating a new watchkv interface for watch usages.
Only datacoord and datanode need watch() , the rest of codebase only needs basic metakv ability.

Issue: https://github.com/milvus-io/milvus/issues/24694

### CODE REVIEW

File: internal/kv/kv.go [hunk 1]
```diff
@@ -67,11 +67,18 @@ type MetaKv interface {
 	TxnKV
 	GetPath(key string) string
 	LoadWithPrefix(key string) ([]string, []string, error)
+	CompareVersionAndSwap(key string, version int64, target string, opts ...clientv3.OpOption) (bool, error)
```
Comment (jiaoew1991) [inline internal/kv/kv.go:70]: As an interface, it's better to remove the parameters related to etcd clientv3 implementation 🤔

Comment (yiwangdr) [reply internal/kv/kv.go]: Done.
Not sure why it is there in the first place as there is no usage. It's probably due to legacy reasons.

Comment (jiaoew1991) [reply internal/kv/kv.go]: Indeed, it is due to historical reasons. The code iteration is very fast and there are many legacy code that need to be cleaned up.

**Pull request merged**